packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -66,57 +67,117 @@ function BlockBindingsPanelMenuContent( { fieldsList, attribute, binding } ) {
 		},
 		[ clientId, attribute ]
 	);
+
 	return (
-		<>
-			{ Object.entries( fieldsList ).map( ( [ name, fields ], i ) => (
-				<Fragment key={ name }>
-					<Menu.Group>
-						{ Object.keys( fieldsList ).length > 1 && (
-							<Menu.GroupLabel>
-								{ registeredSources[ name ].label }
-							</Menu.GroupLabel>
-						) }
-						{ Object.entries( fields )
-							.filter(
-								( [ , args ] ) => args?.type === attributeType
-							)
-							.map( ( [ key, args ] ) => (
-								<Menu.RadioItem
-									key={ key }
-									onChange={ () =>
-										updateBlockBindings( {
-											[ attribute ]: {
-												source: name,
-												args: { key },
-											},
-										} )
-									}
-									name={ attribute + '-binding' }
-									value={ key }
-									checked={ key === currentKey }
-								>
-									<Menu.ItemLabel>
-										{ args?.label }
-									</Menu.ItemLabel>
-									<Menu.ItemHelpText>
-										{ args?.value }
-									</Menu.ItemHelpText>
-								</Menu.RadioItem>
-							) ) }
-					</Menu.Group>
-					{ i !== Object.keys( fieldsList ).length - 1 && (
-						<Menu.Separator />
-					) }
-				</Fragment>
-			) ) }
-		</>
+		<Menu placement={ isMobile ? 'bottom-start' : 'left-start' }>
+			{ Object.entries( sources ).map( ( [ sourceKey, source ] ) => {
+				// Only show sources that have compatible data for this specific attribute.
+				const sourceDataItems = source.data?.filter(
+					( item ) => item?.type === attributeType
+				);
+
+				// Skip sources with no compatible data
+				if ( ! sourceDataItems || sourceDataItems.length === 0 ) {
+					return null;
+				}
+
+				if ( source.mode === 'dropdown' ) {
+					return (
+						<Menu
+							key={ sourceKey }
+							placement={
+								isMobile ? 'bottom-start' : 'left-start'
+							}
+						>
+							<Menu.SubmenuTriggerItem>
+								<Menu.ItemLabel>
+									{ source.label }
+								</Menu.ItemLabel>
+							</Menu.SubmenuTriggerItem>
+							<Menu.Popover gutter={ 8 }>
+								<Menu.Group>
+									{ sourceDataItems.map( ( item ) => (
+										<Menu.CheckboxItem
+											key={ item.key }
+											onChange={ () => {
+												const isCurrentlySelected =
+													source.isSelected?.( {
+														item,
+														binding,
+													} ) ||
```

Comment:
bug here: returning `false` from `isSelected` fails the `||` operator
```suggestion
													} ) ??
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/71542/commits/1f01c3d80f398b3fa3bd5b6dc89e0e301d75f23b